Compile error when cross-compiling to `armv7-linux-androideabi`.

```text
error[E0308]: mismatched types
   --> D:\AppData\Languages\Cargo\registry\src\index.crates.io-6f17d22bba15001f\atomic-write-file-0.2.1\src\imp\unix\mod.rs:249:43
    |
249 |         let mode = Mode::from_bits_retain(stat.st_mode);
    |                    ---------------------- ^^^^^^^^^^^^ expected `u16`, found `u32`
    |                    |
    |                    arguments to this function are incorrect
    |
note: associated function defined here
   --> D:\AppData\Languages\Cargo\registry\src\index.crates.io-6f17d22bba15001f\nix-0.29.0\src\sys\stat.rs:29:1
    |
29  | / libc_bitflags! {
30  | |     /// "File mode / permissions" flags.
31  | |     pub struct Mode: mode_t {
32  | |         /// Read, write and execute for owner.
...   |
61  | |     }
62  | | }
    | |_^
    = note: this error originates in the macro `$crate::__impl_bitflags` which comes from the expansion of the macro `libc_bitflags` (in Nightly builds, run with -Z macro-backtrace for more info)
help: you can convert a `u32` to a `u16` and panic if the converted value doesn't fit
    |
249 |         let mode = Mode::from_bits_retain(stat.st_mode.try_into().unwrap());
    |                                                       ++++++++++++++++++++

For more information about this error, try `rustc --explain E0308`.
error: could not compile `atomic-write-file` (lib) due to 1 previous error
```

--- a/src/imp/unix/mod.rs
+++ b/src/imp/unix/mod.rs
@@ -246,7 +246,8 @@ fn copy_file_perms<P: AsRef<Path>>(
         Err(err) => return Err(err.into()),
     };
     if opts.preserve_mode {
-        let mode = Mode::from_bits_retain(stat.st_mode);
+        #[allow(clippy::unnecessary_cast)]
+        let mode = Mode::from_bits_retain(stat.st_mode as mode_t);
         fchmod(copy_to.as_raw_fd(), mode)?;
     }
     if opts.preserve_owner.is_yes() {

